primitives/io/src/lib.rs

```diff
@@ -175,13 +187,60 @@ pub trait Storage {
 	/// backend.
 	#[version(2)]
 	fn clear_prefix(&mut self, prefix: &[u8], limit: Option<u32>) -> KillStorageResult {
-		let (all_removed, num_removed) = Externalities::clear_prefix(*self, prefix, limit);
-		match all_removed {
-			true => KillStorageResult::AllRemoved(num_removed),
-			false => KillStorageResult::SomeRemaining(num_removed),
+		let r = Externalities::clear_prefix(*self, prefix, limit, None);
+		match r.maybe_cursor {
+			None => KillStorageResult::AllRemoved(r.loops),
+			Some(_) => KillStorageResult::SomeRemaining(r.loops),
```

Comment:
If we're returning the number of keys removed from the backend, then these two should be `r.backend`.

Reply:
You mean we used to return the number of loops done instead of the number of keys that were deleted? If so, the documentation on `KillStorageResult` must be updated to match this behaviour.

Reply:
yes if you look at master that's what is done.

Updating documentation can be good yes (just looked in spec.polkadot.network it is just written removed value).